src/imputors/knn.jl

```diff
@@ -0,0 +1,79 @@
+"""
+    KNN <: Imputor
+
+Imputation using k-Nearest Neighbor algorithm.
+
+# Keyword Arguments
+* `num_nn::Int`: number of nearest neighbors
+* `dist::MinkowskiMetric`: distance metric suppports by `NearestNeighbors.jl` (Euclidean, Chebyshev, Minkowski and Cityblock)
+* `on_complete::Function`: a function to run when imputation is complete
+
+# Reference
+* Troyanskaya, Olga, et al. "Missing value estimation methods for DNA microarrays." Bioinformatics 17.6 (2001): 520-525.
+"""
+# TODO : Support Categorical Distance (NearestNeighbors.jl support needed)
+struct KNN{M} <: Imputor where M <: NearestNeighbors.MinkowskiMetric
+    num_nn::Int
+    dist::M
+    context::AbstractContext
+end
+
+function KNN(; num_nn=1,
+    dist=Euclidean(), context=Context())
+
+    KNN(num_nn, dist, context)
+end
+
+"""
+    impute!(imp::KNN, data::AbstractMatrix)
+
+data : N x D matrix
+"""
```

Comment:
Probably don't actually need this doc string as it's already covered by the more general `impute!` docstrings.

Reply:
I removed docstring